Custom Event Refactor `[AARD-2080]`

rutmanz opened the pull request:
## Task

<!--
Please include any relevant Jira ticket ID(s) at the end of the PR title, in the form AARD-xxxx, where "AARD" is Jira project.
Include the same Jira ticket ID(s) in this section.
-->

AARD-2080

<!--
Provide a brief description of what the task was here.
-->

## Symptom
<!--
How does the problem manifest itself?
Describe the problem as seen by the user, or by the caller or the code if it's not directly visible to the user.

Note: "Symptom" can include new product use cases, not just bugs.
-->
Currently there are a bunch of custom event classes with inconsistent implementations that both make it harder to find events and make the whole system more verbose. Unsubscription is also often done incorrectly or not done at all

## Solution

<!--
How did you fix the problem/symptom?
Explain your approach and reasoning for choosing this solution.
-->

I introduced an `EventSystem` class that manages all custom events and has generic `create`, `dispatch`, and `listen` methods. 

## Verification

<!--
How did you test and verify your changes were correct?
List steps taken, tests/added/updated, and any manual verification done that should be replicated in review.
-->

Go through and try and trigger the list of events found in `EventDataMap` and confirm that they still work as before. This PR should not affect functionality

---

Before merging, ensure the following criteria are met:

- [x] All acceptance criteria outlined in the ticket are met.
- [x] Necessary test cases have been added and updated.
- [x] A feature toggle or safe disable path has been added (if applicable).
- [x] User-facing polish:
  - Ask: *"Is this ready-looking?"*
- [x] Cross-linking between Jira and GitHub:
  - PR links to the relevant Jira issue.
  - Jira ticket has a comment referencing this PR.

rutmanz committed:
feat: replace all event subclasses with EventSystem

rutmanz committed:
fix: unit tests

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

rutmanz committed:
fix: move inputschemechanged event to new system

rutmanz commented:
(because other open PRs add new events in ways that won't necessarily show up as conflicts, I'll want to do one last check before this is merged)

rutmanz committed:
fix: run formatter

rutmanz committed:
fix: remove debugging tools

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

AlexD717 reviewed (approved):
lgtm

azaleacolburn commented on fission/src/mirabuf/ProtectedZoneSceneObject.ts:
```diff
@@ -136,21 +136,22 @@ class ProtectedZoneSceneObject extends SceneObject {
+                        const body1 = message.GetBody1ID()
+                        const body2 = message.GetBody2ID()
+
+                        if (body1.GetIndexAndSequenceNumber() == this._joltBodyId?.GetIndexAndSequenceNumber()) {
```
This condition appears a lot, could we perhaps extract it into a function like `bodyEq(body1: Jolt.BodyID, body2: Jolt.BodyID): bool`.

rutmanz replied:
not sure why that would be better or in scope?

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

BrandonPacewic reviewed (changes requested):
There are some non-trival conflicts here. Since this is not user facing this will not be put in for `v7.2`.

rutmanz committed:
Merge remote-tracking branch 'origin/dev' into zachr/2080/events

rutmanz committed:
fix: run formatter

rutmanz commented:
Just merged `dev` and dealt with the conflicts

BrandonPacewic committed:
fix: cleanups + prevent test listeners persist on window

BrandonPacewic committed:
chore: merge `dev`

* dev: (44 commits)
  fix: don't add new zone to array before navigating to config
  fix: revert `fission/src/util/TypeConversions.ts` formatting
  fix: biome fmt issues
  fix: build isuses
  fix: correct zone refactor regressions
  fix: biome lint warnings
  feat: import unstable mira files with warnings
  revert: `2fd4c67aae0be43b473ed0e5b5f2d90f17ecfcf6`
  feat: handle zero triangles in `createConcaveShapeSettingsFromPart`
  doc: add current status to in development features
  docs: update manual exporter install tutorial
  docs: tutorial content updates
  fix: spelling, grammer, and formatting corrections
  Update tutorials/CommandPalette.md
  Update tutorials/SpawnAsset.md
  Update tutorials/SpawnAsset.md
  Update tutorials/Multiplayer.md
  Update tutorials/MatchMode.md
  Update tutorials/MatchMode.md
  Update tutorials/MatchMode.md
  ...

BrandonPacewic committed:
fix: remove dead entries in `EventDataMap`

BrandonPacewic reviewed (approved):
I went through and double checked changes to events made pre and post this PR. I believe we should be good to go. If anything else comes up I'll open another PR.

Thanks for doing this. Love me a good refactor!
